db/db_impl/db_impl.cc

```diff
@@ -4726,8 +4727,35 @@ Status DBImpl::CreateColumnFamilyWithImport(
   return status;
 }
 
+Status DBImpl::VerifyFileChecksums(const ReadOptions& read_options) {
+  return VerifyChecksum(read_options, /*use_file_checksum=*/true);
+}
+
 Status DBImpl::VerifyChecksum(const ReadOptions& read_options) {
+  return VerifyChecksum(read_options, /*use_file_checksum=*/false);
+}
+
+Status DBImpl::VerifyChecksum(const ReadOptions& read_options,
+                              bool use_file_checksum) {
   Status s;
+
+  std::string file_checksum_func_name(kUnknownFileChecksumFuncName);
+  if (use_file_checksum) {
+    FileChecksumGenFactory* const file_checksum_gen_factory =
+        immutable_db_options_.file_checksum_gen_factory.get();
+    if (!file_checksum_gen_factory) {
+      s = Status::InvalidArgument(
+          "use_file_checksum cannot be true if file checksum is disabled");
```

Comment:
use_file_checksum is not part of the public API, so doesn't make sense in a Status result passed back to the user.

Reply:
Sorry, this comes from a previous version. Will update.